deployment/ops/mcms/op_proposal_generate.go

```diff
@@ -73,20 +102,10 @@ var generateProposalHandler = func(b cld_ops.Bundle, deps sui_ops.OpTxDeps, inpu
 		if err != nil {
 			return mcms.TimelockProposal{}, fmt.Errorf("failed to execute operation %s: %w", def.ID, err)
 		}
-		// Use reflection to extract the Call field from the OpTxResult regardless of its specific type
-		outputValue := reflect.ValueOf(res.Output)
-		if outputValue.Kind() != reflect.Struct {
-			return mcms.TimelockProposal{}, fmt.Errorf("operation %s did not return a struct output", def.ID)
-		}
-
-		callField := outputValue.FieldByName("Call")
-		if !callField.IsValid() {
-			return mcms.TimelockProposal{}, fmt.Errorf("operation %s output does not have a Call field", def.ID)
-		}
-
-		call, ok := callField.Interface().(sui_ops.TransactionCall)
-		if !ok {
-			return mcms.TimelockProposal{}, fmt.Errorf("operation %s Call field is not a TransactionCall", def.ID)
+		// Extract the Call field
+		call, err := extractTransactionCall(res.Output, def.ID)
```

Comment:
It was failing in cases where the operation was in the Op Report storage, refactored it